### PR TITLE
compat(node/http): polyfill `ClientRequest.setTimeout()`

### LINKED ISSUE #17852: compat(node/https): ClientRequest.setTimeout is not a function

### PR DESCRIPTION
**Pull request by schwarzkopfb**
Add a simple implementation of `ClientRequest.setTimeout()` that simulates Node's behaviour. Fixes #17852 and eliminates the excess warning messages ("not implemented...") from stdout that appeared in #16810.

### CODE REVIEW

File: ext/node/polyfills/http.ts [hunk 1]
```diff
@@ -355,8 +356,18 @@ class ClientRequest extends NodeWritable {
...
-  setTimeout() {
-    console.log("not implemented: ClientRequest.setTimeout");
+  // It doesn't relies on the underlying socket (as in Node.js),
+  // we only have a dummy socket here anyway
+  // but it's required by some libraries so we just simulate it
+  setTimeout(timeout: number, callback?: () => void) {
+    if (typeof callback === "function") {
+      this.once("timeout", callback);
+    }
+    if (this.#timeout) clearTimeout(this.#timeout);
```
Comment (bartlomieju) [inline ext/node/polyfills/http.ts:366]: This looks okay - but I suggest you use `AbortController` and pass `signal` option to `fetch` call created in `_final` - that way you will be able to properly cleanup all resources and ensure there are no funky race conditions going on (it is my understanding that after firing `timeout` event response might still fire another event.

**Review by bartlomieju — COMMENTED** [review]
Looks good! Thanks for the PR @schwarzkopfb. It would be great to enable some of the Node.js tests to ensure that this functionality works as expected. @kt3k could you advise here?

**Review by aapoalas — CHANGES_REQUESTED** [review]
As @bartlomieju said, we'll need to find a test to enable to go along with this, and using `AbortController` sounds like a good idea.

**Comment by levex** [comment]
I think we can close this given that #18783 was merged, @bartlomieju @schwarzkopfb ?

**Comment by bartlomieju** [comment]
Yeah! Thanks for the PR @schwarzkopfb, but gonna close without merge.

**Pull request closed without merging**